File: latihan2/Human.java
package DPBO.latihan2;

class Human{
    private String NIK;
    private String nama;
    private String jenis_kelamin;

    public Human() {
        this.NIK = "";
        this.nama = "";
        this.jenis_kelamin = "";
    }

    public Human(String NIK, String nama, String jenis_kelamin) {
        this.NIK = NIK;
        this.nama = nama;
        this.jenis_kelamin = jenis_kelamin;
    }

    public String getNIK(){
        return NIK;
    }

    public void setNIK(String NIK){
        this.NIK = NIK;
    }

    public String getNama(){
        return nama;
    }

    public void setNama(String nama){
        this.nama = nama;
    }

    public String getJenisKelamin(){
        return jenis_kelamin;
    }

    public void setJenisKelamin(String jenis_kelamin){
        this.jenis_kelamin = jenis_kelamin;
    }
}

File: C++/Human.cpp
#include <iostream>
#include <string>

using namespace std;

//superclass
class Human {
   private:
    string NIK;
    string nama;
    char jenis_kelamin;

   public:
    //constructor
    Human(string NIK, string nama, char jenis_kelamin) {
        this->NIK = NIK;
        this->nama = nama;
        this->jenis_kelamin = jenis_kelamin;
    }

    //constructor
    Human() {
        this->NIK = "";
        this->nama = "";
        this->jenis_kelamin = '-';
    }

    //set NIK
    void setNIK(string NIK){
        this->NIK = NIK;
    }

    //get NIK
    string getNIK() {
        return NIK;
    }

    //set nama
    void setNama(string nama){
        this->nama = nama;
    }

    //get nama
    string getNama() {
        return nama;
    }

    //set jenis kelamin
    void setJenisKelamin(char jenis_kelamin){
        this->jenis_kelamin = jenis_kelamin;
    }

    //get jenis kelamin
    char getJenisKelamin() {
        return jenis_kelamin;
    }
};